libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -135,50 +132,30 @@ - (NSError *)errorWithObject:(id)obj withDomain:(NSString *)domain {
 - (NSRegularExpression *)entireRegularExpressionWithPattern:(NSString *)regexPattern
                                                     options:(NSRegularExpressionOptions)options
                                                       error:(NSError **)error {
-  [_entireStringCacheLock lock];
-
-  @try {
```

Comment:
Why was this even here in the first place?

Reply:
To make sure the unlock happened if an invalid regex pattern sneaks in and throws an exception. Now we're trusting that NSCache handles that correctly.